### PR TITLE
Issue #SB-25924 feat: Projects with Invite Only option for contributions

### PR DESCRIPTION
**Pull request by ahghatol**
<table>
<td>
  <a href="https://project-sunbird.atlassian.net/browse/SB-25924" title="SB-25924" target="_blank">StorySB-25924</a>  Projects with "Invite Only" option for contributions - Part 1
  </td></table>
  <br />
 

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

**Test Configuration**:
* Software versions: Nodejs 8.11.2
* Hardware versions:

### Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/helpers/programHelper.js [hunk 1]
```diff
@@ -1184,12 +1186,60 @@ class ProgramServiceHelper {
...
   }
+
+
+  /**
+   * Is user already nominated
+   * @param {*} program_id  program id
+   * @param {*} orgosid     Open saber org id
+   * @returns boolean
+   */
+  async isAlreadyNominated(program_id, orgosid) {
```
Comment (vaivk369) [inline src/helpers/programHelper.js:1203]: @ahghatol  can you make this generalised for org as well as indidual?

Comment (ahghatol) [reply src/helpers/programHelper.js]: Done

File: src/service/programService.js [hunk 2]
```diff
@@ -3190,6 +3191,80 @@ function getParams(msgId, status, errCode, msg) {
...
+          user_id: _.get(program, 'createdby')
+        }
+      });
+
+      const collection_ids = _.get(nomination, 'collection_ids') || [];
+      return await nominateRestrictedContributors(req, program, collection_ids);
+    }
+  }
+  catch(err) {
+    console.log(err);
```
Comment (pallakartheekreddy) [inline src/service/programService.js:3219]: Please improve log

File: src/service/programService.js [hunk 3]
```diff
@@ -3190,6 +3191,80 @@ function getParams(msgId, status, errCode, msg) {
...
+            usersToNotify.push(org.User);
+          }
+        }
+
+        return await programServiceHelper.notifyRestrictedContributors(req, program, usersToNotify);
+      }
+    }
+  }
+  catch (err) {
+    console.log(err);
```
Comment (pallakartheekreddy) [inline src/service/programService.js:3260]: Same here

**Review by pallakartheekreddy — CHANGES_REQUESTED** [review]

**Review by pallakartheekreddy — APPROVED** [review]

**Pull request merged**